### ISSUE TITLE
Quarkus Maven Plugin doubles up the /target folder when used with Flatten Plugin

### ISSUE THREAD

**Issue by kjq** [quarkusio/quarkus#34764]
### Describe the bug

Upgrading to 3.2.0.Final the Quarkus Maven Plugin is doubling up the `/target` folder.  Downgrading it to 3.1.3 does not have this issue.

Running this:
```
mvn quarkus:dev -Dquarkus.analytics.disabled=true
```

Results in:
```
Hot reloadable dependency i.. has not been compiled yet (the classes directory /home/.../target/target/classes does not exist)
```

### Expected behavior

Target folder is treated as is (was in 3.13 and before)

### Actual behavior

An additional `/target` folder looks to be appended

### How to Reproduce?

Running `mvn quarkus:dev -Dquarkus.analytics.disabled=true`

### Quarkus version or git rev

3.2.0.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Maven 3.9.1

### Additional information

I cannot just downgrade the plugin because it affect creating the artifacts with 3.2

**Comment by gastaldi** [comment]
Can you provide a reproducer?

**Comment by kjq** [comment]
Ill try but im not sure how yet.  I do know using 3.1.3 works and using 3.2.0 does not.  Im not sure if it is a maven thing (though we have not update that in a while i believe).

**Comment by gastaldi** [comment]
Are you testing against a multimodule project?

**Comment by kjq** [comment]
Yes

**Comment by gastaldi** [comment]
I have a small multimodule project here: https://github.com/gastaldi/quarkus-multi-module-project-quickstart

Can you change it to reproduce your use case?

**Comment by kjq** [comment]
Trying to replicate it with a multi-module on my end still.

**Comment by kjq** [comment]
Interestingly enough, if i do `mvn quarkus:dev -am -pl bootstrap -Dquarkus.analytics.disabled=true` it starts up but it seems to go through each dependency as if it were standalone before coming up.

**Comment by kjq** [comment]
I seem to have "sort of" a worksaround - doing `mvn quarkus:dev -pl bootstrap -Dquarkus.analytics.disabled=true` works but it does not seem to respect the application.yaml in the bootstrap.

**Comment by kjq** [comment]
Just came across this as well: `https://github.com/quarkusio/quarkus/issues/33725` - will try it to see

**Comment by kjq** [comment]
I've replicated it!  It may have something to do with Maven "version" friendly builds and the combination of the Maven Flatten plugin.

[code-with-quarkus.zip](https://github.com/quarkusio/quarkus/files/12053709/code-with-quarkus.zip)

Steps to reproduce:
1. Unzip the example
2. Run `mvn install` in the parent folder
3. `cd projectc`
4. `mvn quarkus:dev -Dquarkus.analytics.disabled=true`

The output:
```
[ERROR] Failed to execute goal io.quarkus.platform:quarkus-maven-plugin:3.2.0.Final:dev (default-cli) on project projectc: Failed to run: Hot reloadable dependency org.acme:projectc:6.1.15-SNAPSHOT has not been compiled yet (the classes directory /home/kjq/work/code-with-quarkus/projectc/target/target/classes does not exist) -> [Help 1]
```

**Comment by gastaldi** [comment]
@kjq I've investigated and noticed that the issue is in our tooling assuming that the project directory is always the parent directory of the processed model. This is true in the majority of use cases, except where you use another plugin to pre-process the pom.xml (flatten-maven-plugin), which creates the pom.xml under `$PROJECT_DIR/target/.flattened-pom.xml`.  
 
See if https://github.com/quarkusio/quarkus/pull/34766 fixes your problem.

**Comment by aloubyansky** [comment]
This is a consequence of invoking all the plugins with phases preceding `compile` and `quarkus:dev`. As a workaround until we fix it, moving the flatten plugin to a post-compile phase should work.
Thanks for the reproducer @kjq and @gastaldi for looking into this. I'll have a look as well. We'll need to this reproducer to our testsuite too.

**Comment by aloubyansky** [comment]
`mvn compile quarkus:dev` fails in 3.1.3.Final too. So, generally, the issue is present in earlier Quarkus versions too, it's just the `quarkus:dev` impl in earlier versions was limited to a specific "prepartion" goals.

**Comment by kjq** [comment]
I will test more but adjusting the phase to `process-classes` seems to work (good call on that).  

If this works and it is a crazy edge case for you to fix then I am content with the workaround.

We don't like having to use the flatten plugin but it is a necessary evil for Maven friendly versions when deploying since we do not use the release plugin.  On one side it makes Maven releases bearable, and in theory sounded great, but applied it has been a different story. 

I will say if we dont sneeze it has worked good and allows us to rapidly change versions, mostly between releases and -SNAPSHOTs, across the board.

```xml
<executions>
    <execution>
        <id>flatten</id>
        <phase>process-classes</phase>
        <goals>
            <goal>flatten</goal>
        </goals>
    </execution>
    <execution>
        <id>flatten.clean</id>
        <phase>clean</phase>
        <goals>
            <goal>clean</goal>
        </goals>
    </execution>
</executions>
```

Odd you mention it did not work in 3.1.3, we have been using it for a few years in conjunction with Quarkus and have not _noticed_ any issues when starting up  using `quarkus:dev`

Our deployed services of course work because at that point we are using a released "final" POM (which is what the flattened pom becomes when we deploy) so it seems this is specific to `quarkus:dev` and nothing else.

**Comment by aloubyansky** [comment]
If you always used `mvn quarkus:dev` you wouldn't run into it with previous versions. It would fail if you added, e.g. `compile` before `quarkus:dev`.
It will be fixed in 3.2.1.Final.
